iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesSensor.java

```diff
@@ -53,21 +54,23 @@
 public class KubernetesSensor extends YamlSensor {
   private static final Logger LOG = LoggerFactory.getLogger(KubernetesSensor.class);
   private static final String HELM_ACTIVATION_KEY = "sonar.kubernetes.internal.helm.enable";
-  private final HelmProcessor helmProcessor;
+  private final HelmEvaluator helmEvaluator;
+
+  private HelmProcessor helmProcessor;
   private final LocationShifter locationShifter = new LocationShifter();
   private final KubernetesParserStatistics kubernetesParserStatistics = new KubernetesParserStatistics();
 
   public KubernetesSensor(SonarRuntime sonarRuntime, FileLinesContextFactory fileLinesContextFactory, CheckFactory checkFactory,
-    NoSonarFilter noSonarFilter, KubernetesLanguage language, HelmProcessor helmProcessor) {
+    NoSonarFilter noSonarFilter, KubernetesLanguage language, HelmEvaluator helmEvaluator) {
```

Comment:
The `KubernetesSensor` should have a dependency on `HelmProcessor` as it is described in our design picture (Lucid Chart).
The `HelmEvaluator` is implementation detail how `HelmProcessor` is implemented and should not be used here. 
Thanks to that, we can avoid having the `setHelmProcessorForTesting` method, which is here only for testing purposes.

Reply:
The KubernetesSensor has a dependency on the processor. It is a field of the sensor.
The evaluator is a dependency if it will be injected. Jonas mentioned that this will be necessary for SonarLint context. IMO this does not contradict the new design. Happy to discuss it further.